src/uniformsort.hpp

```diff
@@ -67,35 +60,35 @@ namespace UniformSort {
             buf_size--;
             // First unique bits in the entry give the expected position of it in the sorted array.
             // We take 'bucket_length' bits starting with the first unique one.
-            uint64_t pos =
-                Util::ExtractNum(buffer.get() + buf_ptr, entry_len, bits_begin, bucket_length) *
-                entry_len;
+            uint64_t idx =
+                Util::ExtractNum(buffer.get() + buf_ptr, entry_len, bits_begin, bucket_length);
+            uint64_t mem_ptr = idx * entry_len;
```

Comment:
I think it's misleading to call this `_ptr`. It's not a pointer, it's an offset

Reply:
I based the naming on `buf_ptr`, which is also an offset instead of a pointer. I agree that it's misleading, I'll change it.